src/error.rs

```diff
@@ -187,6 +204,7 @@ pub enum Error {
     /// Riker system error during Stronghold initialization
     #[cfg(feature = "stronghold")]
     #[error("Stronghold reported a system error: {0}")]
```

Comment:
I've decided to use `Display` trait for serializing error types that don't implement Serialize, the only slight downside here is that it ignores the string message generated by the `#[error()]` macro which I think is fine.